botorch/models/higher_order_gp.py

```diff
@@ -659,7 +663,7 @@ def posterior(
                 covar = cm(param)
                 if covar.batch_shape != batch_shape:
                     covar = BatchRepeatLazyTensor(covar, batch_shape)
-                joint_covar_list.append(covar)
+                joint_covar_list.append(covar.detach())
```

Comment:
why not move this up and do `covar = cm(param).detach()`? That would avoid calling this on a `LazyTensor`